src/TextArea.tsx

```diff
@@ -0,0 +1,164 @@
+  min-height: 124px;
+  font-family: inherit;
+  font-size: ${get('fontSizes.1')};
+  line-height: 20px;
+  color: ${get('colors.fg.default')};
+  vertical-align: middle;
+  background-color: ${get('colors.canvas.default')};
+  border: 1px solid ${get('colors.border.default')};
+  border-radius: ${get('radii.2')};
+  outline: none;
+  box-shadow: ${get('shadows.primer.shadow.inset')};
+  cursor: text;
+  display: inline-flex;
+  align-items: stretch;
+  padding: ${get('space.2')};
+
+  &:focus-within {
+    border-color: ${get('colors.accent.emphasis')};
+    box-shadow: ${get('shadows.primer.shadow.focus')};
+  }
+
+  &::placeholder {
+    color: ${get('colors.fg.subtle')};
+  }
+
+  ${props =>
+    props.validationStatus === 'error' &&
+    css`
+      border-color: ${get('colors.danger.emphasis')};
+      &:focus-within {
+        border-color: ${get('colors.danger.emphasis')};
+        box-shadow: ${get('shadows.btn.danger.focusShadow')};
+      }
+    `}
+
+  ${props =>
+    props.validationStatus === 'success' &&
+    css`
+      border-color: ${get('colors.success.emphasis')};
+      &:focus-within {
+        border-color: ${get('colors.success.emphasis')};
+        box-shadow: 0 0 0 3px ${get('colors.success.muted')};
+      }
+    `}
+  
+    ${props =>
+    props.block &&
+    css`
+      width: 100%;
+      display: flex;
+    `}
+
+  
+
+  ${props =>
+    props.resize &&
+    css`
+      resize: ${props.resize};
+    `}
+
+    ${props =>
+    props.width &&
+    css`
+      resize: ${props.resize === 'vertical' && !props.height ? 'vertical' : 'none'};
+      width: ${props.width}px;
+    `} 
+
+    ${props =>
+    props.height &&
+    css`
+      resize: ${props.resize === 'horizontal' && !props.width ? 'horizontal' : 'none'};
+      height: ${props.height}px;
+    `} 
+
+    ${props =>
+    props.disabled &&
+    css`
+      resize: none;
+      color: ${get('colors.primer.fg.disabled')};
+      background-color: ${get('colors.input.disabledBg')};
+      border-color: ${get('colors.border.default')};
+      cursor: not-allowed;
+    `}
+
+  @media (min-width: 360px) {
+    min-width: 300px;
+  }
+
+  // Ensures inputs don't zoom on mobile but are body-font size on desktop
+  @media (min-width: ${get('breakpoints.1')}) {
+    font-size: ${get('fontSizes.1')};
+  }
+
+  ${sx}
+`
+
+/**
+ * An accessible, native textarea component that supports validation states.
+ * This component accepts all native HTML <textarea> attributes as props.
+ */
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  (
+    {disabled, sx: sxProp, required, validationStatus, width, height, resize = 'both', ...rest}: TextAreaProps,
+    ref
+  ): ReactElement => {
+    return (
+      <StyledTextArea
+        ref={ref}
+        disabled={disabled}
+        sx={sxProp}
+        validationStatus={validationStatus}
+        resize={resize}
+        width={width}
+        height={height}
+        aria-required={required ? 'true' : 'false'}
```

Comment:
What's the difference between `aria-required` and [`required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-required)?

Reply:
I think they have different levels of screenreader support. I think `aria-required` is the preferred one. I bet somebody on the accessibility team would know.

cc @alliethu

Reply:
This was an oversight on my part, I was actually meant to include `required` too. I don't think it hurts to add both, but I have seen some data indicating `aria-*` offers improved support too.

Reply:
@rezrah has it right. It's suggested to use both for wider browser/screen reader support. :)